osfoffline/filesystem_manager/osf_event_handler.py

```diff
@@ -203,18 +204,18 @@ def on_deleted(self, event):
         # get item
         item = self._get_item_by_path(src_path)
 
-        # put item in delete state
-        item.locally_deleted = True
-
-        # nodes cannot be deleted online. THUS, delete it inside database. It will be recreated locally.
-        if isinstance(item, Node):
-            session.delete(item)
-            save(session)
-            return
-
-        save(session, item)
-
-        logging.info('{} set to be deleted'.format(src_path.full_path))
+        # put item in delete state after waiting a second and
+        # checking to make sure the file was actually deleted
+        yield from asyncio.sleep(1)
+        if not os.path.exists(item.path):
+            item.locally_deleted = True
+            # nodes cannot be deleted online. THUS, delete it inside database. It will be recreated locally.
+            if isinstance(item, Node):
```

Comment:
Note for regression testing: check whether files inside a project are correctly recreated as well. (there may be DB conflicts when files with the same ID are re-downloaded)

Presumably, there would be no DB cruft because all DB entries referencing files inside the deleted project folder will be cleared from the DB (because the nodes.files relationship is set to cascade deletes).

However, SQLAlchemy docs suggest that may not be enough:
http://docs.sqlalchemy.org/en/rel_0_9/orm/cascades.html
```
The all symbol is a synonym for save-update, merge, refresh-expire, expunge, delete, and using it in conjunction with delete-orphan indicates that the child object should follow along with its parent in all cases, and be deleted once it is no longer associated with that parent.
```